File: algorithms/python/5.py
class Solution:
    def longestPalindrome(self, s):
        """
        :type s: str
        :rtype: str
        """
        if s == '':
            return ''

        n = len(s)
        res = s[0]

        for i in range(n):
            l, r = i, i
            while l-1 >= 0 and r+1 < n and s[l-1]==s[r+1]:
                l -= 1
                r += 1
            if r-l+1 > len(res):
                res = s[l:r+1]

        for i in range(n-1):
            if s[i] != s[i+1]:
                continue
            l, r = i, i+1
            while l-1 >= 0 and r+1 < n and s[l-1]==s[r+1]:
                l -= 1
                r += 1
            if r-l+1 > len(res):
                res = s[l:r+1]

        return res

File: algorithms/rust/5.rs
impl Solution {
    #[allow(dead_code)]
    pub fn longest_palindrome(s: String) -> String {
        let sb = s.as_bytes();
        let (mut left, mut right) = (0, 0);
        for c in 0..s.len() {
            let (mut l, mut r) = (c, c);
            while l >= 1 && r < s.len() - 1 && sb[l - 1] == sb[r + 1] {
                l -= 1;
                r += 1;
            }
            if r - l > right - left {
                left = l;
                right = r;
            }
        }
        for c in 1..s.len() {
            if sb[c - 1] != sb[c] {
                continue;
            }
            let (mut l, mut r) = (c - 1, c);
            while l >= 1 && r < s.len() - 1 && sb[l - 1] == sb[r + 1] {
                l -= 1;
                r += 1;
            }
            if r - l > right - left {
                left = l;
                right = r;
            }
        }
        s[left..=right].to_string()
    }
}

struct Solution;

fn main() {
    println!("Hello, world!");
}
